File: backend/models/kpi.py
from pydantic import BaseModel

class Kpis(BaseModel):
    """Pydantic model for the calculated KPIs."""
    date: str
    mean_glucose: float
    std_dev: float
    cv: float
    tir_percent: float
    tir_accepted: bool
    tar_percent: float
    tbr_percent: float
    titr_percent: float
    titr_accepted: bool
    ea1c: float
    total_readings: int
    total_readings_accepted: bool

File: frontend/src/models/kpi.ts
export interface Kpis {
  date: string;
  mean_glucose: number;
  std_dev: number;
  cv: number;
  tir_percent: number;
  tir_accepted: boolean;
  tar_percent: number;
  tbr_percent: number;
  titr_percent: number;
  titr_accepted: boolean;
  ea1c: number;
  total_readings: number;
  total_readings_accepted: boolean;
}
